compute_tools/src/compute.rs

```diff
@@ -343,4 +346,67 @@ impl ComputeNode {
         self.prepare_pgdata()?;
         self.run()
     }
+
+    // Look for core dumps and collect backtraces.
+    //
+    // EKS worker nodes have following core dump settings:
+    //   /proc/sys/kernel/core_pattern -> core
+    //   /proc/sys/kernel/core_uses_pid -> 1
+    //   ulimint -c -> unlimited
+    // which results in core dumps being written to postgres data directory as core.<pid>.
+    //
+    // Use that as a default location and pattern, except macos where core dumps are written
+    // to /cores/ directory by default.
+    fn check_for_core_dumps(&self) -> Result<()> {
+        let core_dump_dir = match std::env::consts::OS {
+            "macos" => Path::new("/cores/"),
+            _ => Path::new(&self.pgdata),
+        };
+
+        // Collect core dump paths if any
+        info!(
+            "checking for core dumps in {}",
+            core_dump_dir.to_string_lossy()
+        );
+        let files = fs::read_dir(core_dump_dir)?;
+        let cores = files.filter_map(|entry| {
+            let entry = entry.ok()?;
+            let _ = entry.file_name().to_str()?.strip_prefix("core.")?;
+            Some(entry.path())
+        });
+
+        // Print backtrace for each core dump
+        for core_path in cores {
+            let core_path = core_path.to_string_lossy();
+
+            warn!("core dump found: {}, collecting backtrace", core_path);
+
+            // Try first with gdb
+            let backtrace = Command::new("gdb")
+                .args(["--batch", "-q", &self.pgbin, &core_path, "-ex", "bt"])
```

Comment:
```rust
Command::new("gdb")
  .args(["--batch", "--quiet", "-ex", "bt", &self.pgbin])
  .arg(core_path) // this might have a different type!
```

Reply:
> // this might have a different type!

hm, how so? Now it is `Cow<str>`, is there any problem with that?

Reply:
This is to say that you don't have to convert Path to anything, because it's possible to construct command using objects of different types.

Reply:
Gotcha, fixed